### PR TITLE
Add an option to StirngUtil.split to stuff remaining text in the last

### PR DESCRIPTION
**Pull request by nh13**
array value

Needed for https://github.com/fulcrumgenomics/fgbio/pull/586

### CODE REVIEW

File: src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala [hunk 1]
```diff
@@ -159,28 +159,31 @@ object StringUtil {
...
+    * values.
     * */
-  def split(line: String, delimiter: Char = '\t', arr: Array[String]): Int = {
+  def split(line: String, delimiter: Char = '\t', arr: Array[String], stuffLast: Boolean = false): Int = {
     val cs    = line.toCharArray
     val len   = cs.length
     var i     = 0
     var count = 0
     var start = 0
-    var end   = 0
```
Comment (nh13) [inline src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala:169]: end isn't actually needed!

File: src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala [hunk 2]
```diff
@@ -159,28 +159,31 @@ object StringUtil {
     *
     * Multiple consecutive delimiters are treated as individual delimiters, delimiting empty fields. E.g. with
     * `delimiter=','` the string `a,b,,,c` will return `["a", "b", "", "", "c"]`.
+    *
+    * If `stuffLast` is true, remaining text is placed into the last value in the array instead of discarding extra
+    * values.
     * */
-  def split(line: String, delimiter: Char = '\t', arr: Array[String]): Int = {
+  def split(line: String, delimiter: Char = '\t', arr: Array[String], stuffLast: Boolean = false): Int = {
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala]: Tim suggests: `concatenateRemaining`.

File: src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala [hunk 3]
```diff
@@ -159,28 +159,31 @@ object StringUtil {
...
-      }
-      else {
-        end += 1
       }
 
       i += 1
     }
+
+    if (stuffLast && i < len) {
+      arr(arr.length-1) += "," + new String(cs, start, len-start)
```
Comment (tfenne) [inline src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala]: Shouldn't this be `delimiter`?

File: src/test/scala/com/fulcrumgenomics/commons/util/StringUtilTest.scala [hunk 4]
```diff
@@ -124,4 +124,15 @@ class StringUtilTest extends UnitSpec {
       }
     }
   }
+
+  it should "split a line and place remaining text in the last value of the array when stuffLast=true" in {
```
Comment (tfenne) [inline src/test/scala/com/fulcrumgenomics/commons/util/StringUtilTest.scala:128]: Maybe just loop this test for delimiter = ',' and '\t' to make sure we don't slip up on the delimiter?

**Review by tfenne — APPROVED** [review]
With one oopsie.

File: src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala (hunk 2 above)
Comment (nh13) [reply src/main/scala/com/fulcrumgenomics/commons/util/StringUtil.scala]: So many letters!

**Pull request merged**